File: repos_estudiantes_TARQUI_tercera_entrega/SIÑANI_QUELALI_IVAN/tarea3/ge5.java
import java.util.Stack;
public class Pila<T> {
    private Stack<T> elementos = new Stack<>();
    public void apilar(T elemento) {
        elementos.push(elemento);
    }
    public T desapilar() {
        return elementos.isEmpty() ? null : elementos.pop();
    }
    public void mostrarPila() {
        System.out.println("Contenido de la pila: " + elementos);
    }

    public static void main(String[] args) {
        Pila<Integer> pilaEnteros = new Pila<>();
        pilaEnteros.apilar(10);
        pilaEnteros.apilar(20);
        pilaEnteros.mostrarPila();
        System.out.println("Elemento desapilado: " + pilaEnteros.desapilar());
        pilaEnteros.mostrarPila();

        Pila<String> pilaCadenas = new Pila<>();
        pilaCadenas.apilar("Hola");
        pilaCadenas.apilar("Mundo");
        pilaCadenas.mostrarPila();
        System.out.println("Elemento desapilado: " + pilaCadenas.desapilar());
        pilaCadenas.mostrarPila();
    }
}
//DIAGRAMA
//Clase Pila<T>
//---------------------------
//- elementos: Stack<T>
//---------------------------
//+ apilar(T elemento): void
//+ desapilar(): T
//+ mostrarPila(): void
//---------------------------
//Clase Main
//---------------------------
//+ main(String[] args): void
//---------------------------

File: repos_estudiantes_TARQUI_tercera_entrega/SIÑANI_QUELALI_IVAN/tarea3/ge5.py
class Pila:
    def __init__(self):
        self.elementos = []

    def apilar(self, elemento):
        self.elementos.append(elemento)

    def desapilar(self):
        return self.elementos.pop() if self.elementos else None

    def mostrar_pila(self):
        print("Contenido de la pila:", self.elementos)

pila_enteros = Pila()
pila_enteros.apilar(10)
pila_enteros.apilar(20)
pila_enteros.mostrar_pila()
print("Elemento desapilado:", pila_enteros.desapilar())
pila_enteros.mostrar_pila()

pila_cadenas = Pila()
pila_cadenas.apilar("Hola")
pila_cadenas.apilar("Mundo")
pila_cadenas.mostrar_pila()
print("Elemento desapilado:", pila_cadenas.desapilar())
pila_cadenas.mostrar_pila()


